ipamutils/utils.go

```diff
@@ -37,18 +37,37 @@ type NetworkToSplit struct {
 // InitNetworks initializes the broad network pool and the granular network pool
 func InitNetworks(defaultAddressPool []*NetworkToSplit) {
 	initNetworksOnce.Do(func() {
-		// error ingnored should never fail
-		PredefinedGranularNetworks, _ = splitNetworks(defaultGranularNetwork)
+		mutex.Lock()
+		// error ignored should never fail
+		PredefinedGlobalScopeDefaultNetworks, _ = splitNetworks(globalScopeDefaultNetworks)
+		mutex.Unlock()
 		if defaultAddressPool == nil {
-			defaultAddressPool = defaultBroadNetwork
+			defaultAddressPool = localScopeDefaultNetworks
 		}
 		var err error
-		if PredefinedBroadNetworks, err = splitNetworks(defaultAddressPool); err != nil {
-			logrus.WithError(err).Error("InitAddressPools failed to initialize the default address pool")
+		if PredefinedLocalScopeDefaultNetworks, err = splitNetworks(defaultAddressPool); err != nil {
+			logrus.WithError(err).Error("InitAddressPools failed to initialize the local scope default address pool")
 		}
 	})
 }
 
+// ConfigGlobalScopeDefaultNetworks initializes the global scope default network pool
+func ConfigGlobalScopeDefaultNetworks(defaultAddressPool []*NetworkToSplit) {
+	mutex.Lock()
+	if defaultAddressPool == nil {
+		defaultAddressPool = globalScopeDefaultNetworks
+	}
+	var err error
+	if PredefinedGlobalScopeDefaultNetworks, err = splitNetworks(defaultAddressPool); err != nil {
+		logrus.WithError(err).Error("InitAddressPools failed to initialize the local scope default address pool")
```

Comment:
don't we want to return the err to the caller?

Reply:
This is kind of Init function for swarmKit.  We dont return error. Ideally this should never fail.  Thats why we just log the info.

Reply:
Hrmm.. I'm not sure that I agree in this case or in the `InitNetworks` case.  But especially in this case.  The reason is that both functions take potentially erroneous parameters that can come from user input.  Even though it should never fail, by not returning an error here we are implicitly forcing the caller to know all the rules for input validation that this library requires.  Clearly the CLI and swarmkit should do some input validation.  But IMHO, they should only have to do enough to ensure that the input is parses correctly.  Things like invalid parameter combinations should really be handled here.   This way, if the design or functionality changes, (e.g. becoming more restrictive or less restrictive) the changes don't *have* to happen in multiple locations in the code for it to validate the input correctly.  I'd also think that the caller should be given the option to panic or abort the code entirely if such an error occurs.

Reply:
I agree with you on caller should get notified.  We dont pass user input in while calling Libnetwork API the way we used to do.  Instead of passing CLI input deep into stack, I was asked to change APIs params such a way that we pass well defined structure. when we receive input from CLI , now the parser does all validation and converts into []IPNet object. So  if there is failure in the user input we reject before it reaches swarmKit or Libnetwork. When we call, Libnetwork Init we dont pass wrong structure from swarmkit. So its protected at CLI and swarmKit. Like I mentioned existing Init APIs also behave in the same way.